src/ReleaseApp/Infrastructure/Shared/Dto/ReleaseGroupDto.php

```diff
@@ -59,6 +59,22 @@ public function __construct(
         $this->hasConflict = $hasConflict;
     }
 
+    /**
+     * @param \ReleaseApp\Infrastructure\Shared\Dto\Collection\ModuleDtoCollection $moduleCollection
+     *
+     * @return self
+     */
+    public function withModuleCollection(ModuleDtoCollection $moduleCollection): self
+    {
+        return new self(
```

Comment:
Why do we need to create a new instance here?
```suggestion
        $this->moduleCollection = $moduleCollection;
        
        return $this;
```

Reply:
Or just setter.